src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -594,8 +594,9 @@ protected DynamoModel(IStartConfiguration config)
                 PreferenceSettings = settings;
                 PreferenceSettings.PropertyChanged += PreferenceSettings_PropertyChanged;
             }
-
-            InitializeInstrumentationLogger();
+            // If user does not agree to GA terms, do not try to launch the client at all
+            if (PreferenceSettings.IsAnalyticsReportingApproved)
```

Comment:
would you mind using braces?

Also, we could write a test for this perhaps which checks that the dlls are not loaded - I will leave that up to you to decide if it's necessary to protect this behavior in the future - but here is an example I think would be good to base the test on.

https://github.com/DynamoDS/Dynamo/blob/1d49129f5fc1f0f80abf94ae9bc04833cd069064/test/System/IntegrationTests/DynamoApplicationTests.cs#L34

we would want to startup a new process and check the modules which are loaded, we can't rely on checking the loaded modules of the test because some previous test may have already loaded the modules.

Reply:
Added braces. I did explore the unit testing side, I find Analytics.Net.Core.dll can only be detected using `AppDomain.CurrentDomain.GetAssemblies()` which requires a way to monitor the appdomain came with the new process. I will try to make it into a serparate PR. I did run self CI on this one and everything is passing. So merging for now and cherry-pick to get a build for customer testing first.